### ISSUE TITLE
[IJPL-162016] Semantic highlighting works incorrectly in Scala

### ISSUE THREAD

**Issue by migdeosan** [izhangzhihao/intellij-rainbow-brackets#2781]
### Have you checked the issues and discussions to ensure there are no duplicates?

Yes

### Are you using JetBrains Client / Code With Me / IntelliJ Remote Development?

No

### Your programming languages

Scala

### Free or paid?

Paid users

### Expected Behavior

In the previous version the plugin works fine as in the image.

![image](https://github.com/user-attachments/assets/82fe23ee-dc28-4bf5-a895-7c821ec9b6ce)

### Current Behavior

But in the last version doesn't work as you can see in this image.
![image](https://github.com/user-attachments/assets/1c7da93e-5858-47a1-aa9a-49e3942dc51b)

Sometimes works for some variables but not for others and it doesn't have any sense.

### Code snippet for reproduce

```shell
/**
  * Method for testing the rainbow bracket plugin
  * @param df DataFrame to test
  * @param spark Spark Session handler
  */
def testingRainbowBracket(df: DataFrame)(implicit spark: SparkSession): Unit = {
  val rainbowBracket = spark.table("db.tb")
  val rainbowBracketColumns = rainbowBracket.columns
  val rainbowBracketDF = df
    .join(rainbowBracket, df("product_code") === rainbowBracket("product_code"), "left_outer")
    .select(rainbowBracketColumns.map(col):_*)
  rainbowBracketDF.show(10)
}
```

### Your Environment

```shell
IntelliJ IDEA 2024.2.2 (Community Edition)
Build #IC-242.22855.74, built on September 18, 2024
Runtime version: 21.0.3+13-b509.15 x86_64 (JCEF 122.1.9)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.lwawt.macosx.LWCToolkit
macOS 14.6.1
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 2048M
Cores: 8
Metal Rendering is ON
Registry:
  ide.experimental.ui=true
  i18n.locale=
Non-Bundled Plugins:
  Key Promoter X (2024.2.0)
  lermitage.intellij.extra.icons (2024.7.2)
  Lombook Plugin (242.22855.74)
  com.wakatime.intellij.plugin (15.0.3)
  gherkin (242.20224.159)
  net.seesharpsoft.intellij.plugins.csv (3.4.0-242)
  izhangzhihao.rainbow.brackets (2024.2.6-241)
  com.github.copilot (1.5.23.6819)
  zielu.gittoolbox (600.0.0+242)
  org.sonarlint.idea (10.9.0.79423)
  MavenRunHelper (4.28.0-IJ2022.2)
  org.intellij.scala (2024.2.25)
Kotlin: 242.22855.74-IJ
```

**Comment by izhangzhihao** [comment]
- Which is your previous version?
- Pls show me your settings page

**Comment by migdeosan** [comment]
- The previews version which I'm using is:
IntelliJ IDEA 2023.3.8 (Community Edition)
Build #IC-233.15619.7, built on September 4, 2024
Runtime version: 17.0.12+1-b1087.25 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.6.1
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 8
Metal Rendering is ON
Registry:
  ide.experimental.ui=true
Non-Bundled Plugins:
  Key Promoter X (2024.1.0)
  lermitage.intellij.extra.icons (2024.7.2)
  com.wakatime.intellij.plugin (15.0.3)
  Lombook Plugin (233.15619.7)
  net.seesharpsoft.intellij.plugins.csv (3.4.0-233)
  gherkin (233.11799.165)
  izhangzhihao.rainbow.brackets (2024.2.6-233)
  com.github.copilot (1.5.23.6819)
  zielu.gittoolbox (600.0.0+233)
  org.sonarlint.idea (10.9.0.79423)
  MavenRunHelper (4.28.0-IJ2022.2)
  org.intellij.scala (2023.3.30)
Kotlin: 233.15619.7-IJ

- My plugin settings
![image](https://github.com/user-attachments/assets/160a1056-485a-4123-ad7b-06742d4deb19)
![image](https://github.com/user-attachments/assets/0e7fe036-8d80-4d02-a3a6-cc3acdbe6bc1)
![image](https://github.com/user-attachments/assets/8e07d217-40a6-4cb1-954d-5cc0cc73b941)
![image](https://github.com/user-attachments/assets/a7a9e17f-f176-4b5e-8d9c-0a825a6c4cd0)
![image](https://github.com/user-attachments/assets/abc33db4-288a-4a8e-a0ab-420bcc277ebc)

**Comment by izhangzhihao** [comment]
Did you mean sometimes it works but sometimes does not ?

I think that related to [IJPL-162016 Semantic highlighting works incorrectly in Scala](https://youtrack.jetbrains.com/issue/IJPL-162016).

NOTE: rainbow variables for scala is reusing the same function build by the Scala plugin.

**Comment by migdeosan** [comment]
Hi, yes, it seems to be the same problem. But if I understood you correctly, there is no solution for this because it is an issue derived from the Scala plugin?

**Comment by izhangzhihao** [comment]
Yes, it depends on the Scala plugin.

**Comment by migdeosan** [comment]
Ok, thanks.
Regards.

**Comment by izhangzhihao** [comment]
Fixed in the Scala plugin build: 2024.2.609 (Nightly) 
Fixed in the Scala plugin build: 2024.3.521 (Nightly)
